### PR TITLE
man pages: write and update about mapping related options

### PR DESCRIPTION
**Pull request by masatake**
@k-takata, I submit the homework of the last week:-) Could you review this PR?

I made many small commits. I split the commits to make myself understand what area I worked.
However, the change is not so large.

### CODE REVIEW

**Comment by masatake** [comment]
I'm very sorry that I asked you to do cheerless task.
I am written a crazy spell-checking system for this project.
I would like to install it first.

**Comment by k-takata** [comment]
No problem.
BTW, I find that `--input-encoding=encoding`, `--input-encoding-<LANG>=encoding` and `--output-encoding=encoding` are missing in `ctags.1.rst.in`.
Not sure there are still any other options not listed in `ctags.1.rst.in`.

**Comment by masatake** [comment]
I added undocumented options to the man page incrementally. In this time I added --map-<LANG>.

**Comment by k-takata** [comment]
I got it.

**Comment by masatake** [comment]
About spell checker  I should not integrate it to "make check". Running locally maybe enough for awhile.

Anyway, it works:

```
[yamato@master]~/var/ctags-github% bash misc/cspell
checking 59fd40671e2a93f6b55f1c673555d99b7356e907...0
checking 9948c7eb5398f4847723abab7d02c2ba572bc943...9
	=>	origina
	=>	orignal
checking 1f7319013bb4ded1fb00d91431a879ded769feb2...10
	=>	exapmle
	=>	originl
	=>	seciton
	=>	sofotware
	=>	speciallized
	=>	tiime
checking a4950796ea748208ae63974468c99c736dabcaf4...0
checking 201e5e774fef84527802113969998bd71b14466d...0
[yamato@master]~/var/ctags-github% echo $?
1
```

This tool checks the output of `git show` of all commits between master...
I will never take your valuable time for spell checking.

**Comment by masatake** [comment]
@k-takata, I updated after spell checking.

File: man/ctags-incompatibilities.7.rst.in [hunk 1]
```diff
@@ -18,10 +18,40 @@ SYNOPSIS
...
+This incompatible change is introduced to deal with the following
+situation:
+
+	* "build.xml" as a source file,
+	* The "Ant" parser declares it handles a file name pattern "build.xml", and
+	* The "XML" parser declares it handles a file extension "xml".
+
+Which parser should be used for parsing "build.xml"?  The assumption
+of Universal-ctags is the user may want to use the "Ant" parser; the
+fie name pattern it declares is more specific than the file extension
```
Comment (k-takata) [inline man/ctags-incompatibilities.7.rst.in]: fie -> file

Comment (masatake) [inline man/ctags-incompatibilities.7.rst.in]: Wow.
It comes from

```
typedef enum {
	F_ASSIGNMENT,
	COUNT_FIELD
} ldScriptFied; // <<
```

Comment (masatake) [inline man/ctags-incompatibilities.7.rst.in]: I found fied is nothing to do with this. Fie is ... fie.

**Comment by masatake** [comment]
Updated.

File: man/ctags-incompatibilities.7.rst.in (hunk 1 above)
Comment (k-takata) [inline man/ctags-incompatibilities.7.rst.in]: It should be Field?

Comment (masatake) [inline man/ctags-incompatibilities.7.rst.in]: Yes. I fixed this in  	0f6569a.
I have to say cspell target  is far from perfect.

File: man/ctags.1.rst.in [hunk 2]
```diff
@@ -679,44 +690,55 @@ built-in language names.
...
+	As either the '+' or '-' is encountered in the *list*, the languages
 	following it are added or removed from the current list, respectively.
 	Thus, it becomes simple to replace the current list with a new one, or
-	to add or remove languages from the current list. The actual list of
+	to add or remove languages from the current list.
+
+	The actual list of
 	files for which tags will be generated depends upon the language
-	extension mapping in effect (see the --langmap option). Note that all
+	extension mapping in effect (see the ``--langmap`` option). Note all
```
Comment (k-takata) [inline man/ctags.1.rst.in]: `that` is removed. Is this intended?

**Comment by masatake** [comment]
@k-takata, thank you.

**Pull request merged**